langserver/definition.go

```diff
@@ -21,13 +21,13 @@ import (
 )
 
 // Definition is required by the protocol.Server interface.
-func (s *server) Definition(ctx context.Context, params *protocol.DefinitionParams) ([]protocol.Location, error) {
+func (s *server) Definition(_ context.Context, params *protocol.DefinitionParams) ([]protocol.Location, error) {
 	location, err := s.cache.Find(&params.TextDocumentPositionParams)
 	if err != nil {
-		return nil, nil
+		return nil, err
```

Comment:
That's incorrect.

See #185 for why.

Reply:
I still don't get why we couldn't return an error saying the document is not there. In the point of view of the cache that's an error.

It should be on the side of the caller to decide what to do with it using a sort of switch case on the error.

Reply:
For explanation: An expired document context means, the document content has changed while processing the request. While that's rare, it shouldn't result in an error. However in that case, an error is raised internally, since the request can't be handled in a sane way. These special errors have to be filtered out, before responding to the client.